common/reflectcommon/structFieldsUpdate_test.go

```diff
@@ -90,7 +104,7 @@ func TestAdaptStructureValueBasedOnPath(t *testing.T) {
 
 		err := AdaptStructureValueBasedOnPath(cfg, path, expectedNewValue)
 
-		require.ErrorContains(t, err, "cannot cast field <invalid uint32> to kind <uint32>")
+		require.Equal(t, err.Error(), "unable to cast value 'invalid uint32' of type <string> to type <uint32>")
```

Comment:
For this I would have to convert all messages to `error` with fmt.Errorf("MESSAGE"). I would say comparing strings is fine here.
`ErrorIs` is suitable for errors from files like `errors.go` and I cannot move all returned errors there because I'm using parameters `unable to cast value '%v' of type <%s> to type <%s>`